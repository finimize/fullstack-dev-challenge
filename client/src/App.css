h1,
h2,
h3,
h4,
h5,
h6 {
  font-family: 'Avenir Next', 'Avenir', 'Lato', 'Helvetica', 'Arial',
    'Sans-Serif';
  font-weight: 600;
}

h1 {
  font-size: 24px;
  line-height: 28px;
  color: #1d1f20;
  text-align: center;
}

h2 {
  font-size: 20px;
  line-height: 24px;
  color: #1d1f20;
  text-align: center;
}

h3 {
  font-size: 16px;
  line-height: 20px;
  color: #1d1f20;
}

h4 {
  font-size: 14px;
  color: #1d1f20;
  line-height: 18px;
}

p {
  font-family: 'Avenir Next', 'Avenir', 'Lato', 'Helvetica', 'Arial',
    'Sans-Serif';
  font-weight: 400;
  font-size: 16px;
  line-height: 20px;
  color: #1d1f20;
}

.fmz-white-font {
  color: #ffffff;
}

.header-banner {
  background-color: #00b2ff;
  padding: 20px;
}

.financial-inputs {
  padding: 20px;
}

.input-label {
  margin-bottom: 4px;
  margin-top: 12px;
}
