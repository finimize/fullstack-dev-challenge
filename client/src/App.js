import React, { Component } from 'react'
import './App.css'
import { FinimizeInterestRateCalculator } from './components/FinimizeInterestRateCalculator'

class App extends Component {
  render() {
    return (
      <div className="App">
        <FinimizeInterestRateCalculator />
      </div>
    )
  }
}

export default App
