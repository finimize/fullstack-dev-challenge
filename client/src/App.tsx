import React from 'react'
import './App.css'
import { ChakraProvider, extendTheme } from '@chakra-ui/react'
import { Container } from '@chakra-ui/react'
import DefaultLayout from './components/layouts/Default'
import LineChart from './components/LineChart'
import theme from './theme'
import TextField from './components/TextField'
import RangeSlider from './components/RangeSlider'

const defaultTheme = extendTheme(theme)

// Note: This is just for example purposes
// should be replaced with real data from the server
const tempData = {
    xAxis: [
        0,
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
        23,
        24,
        25,
        26,
        27,
        28,
        29,
        30,
        31,
        32,
        33,
        34,
        35,
        36,
        37,
        38,
        39,
        40,
        41,
        42,
        43,
        44,
        45,
        46,
        47,
        48,
        49,
        50,
    ],
    yAxis: [
        100,
        150,
        180,
        210,
        240,
        350,
        360,
        370,
        380,
        390,
        391,
        392,
        392,
        395,
        395,
        396,
        400,
        404,
        407,
        409,
        410,
        411,
        413,
        414,
        418,
        420,
        423,
        425,
        427,
        430,
        431,
        432,
        435,
        437,
        439,
        492,
        500,
        501,
        510,
        511,
        512,
        524,
        526,
        600,
        700,
        900,
        910,
        911,
        1000,
        1050,
    ],
}

function App() {
    return (
        <ChakraProvider theme={defaultTheme}>
            {/* We've just bundled everything into one file here to
            get you started!*/}
            <DefaultLayout>
                <Container pt={6}>
                    <LineChart
                        title="Savings Over time"
                        xAxisData={tempData.xAxis}
                        yAxisData={tempData.yAxis}
                        xLabel="Years"
                        yLabel="Amount"
                    />
                    <TextField
                        setText={() => {}}
                        type={'text'}
                        timeoutValue={200}
                        placeholder={'Initial Deposit'}
                    />
                    <RangeSlider min={0} max={150} setValue={() => {}} defaultValue={0} />
                    <RangeSlider min={0} max={100} setValue={() => {}} defaultValue={0} />
                </Container>
            </DefaultLayout>
        </ChakraProvider>
    )
}

export default App
