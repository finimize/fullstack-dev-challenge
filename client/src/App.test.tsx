import React from 'react'
import { screen } from '@testing-library/react'
import { renderForTest, mockLineChart } from './test'
import App from './App'

jest.mock('./components/LineChart')

test('renders successfully and shows title', () => {
    mockLineChart()

    renderForTest(<App />)

    expect(screen.getByText('Compound interest projections')).toBeInTheDocument()
})
