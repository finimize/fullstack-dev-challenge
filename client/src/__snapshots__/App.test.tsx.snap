// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders learn react link 1`] = `
<div>
  <div
    class="css-2e097b"
  >
    <div
      class="css-1tmkavh"
    >
      <img
        alt="Finimize"
        class="chakra-image__placeholder css-u4ybgy"
      />
    </div>
    <div
      class="css-yf46qw"
    >
      <div
        class="css-1rb18jj"
      >
        <button
          class="chakra-button css-10dvod4"
          data-testid="step-button-0"
          type="button"
        >
          0
        </button>
        <button
          class="chakra-button css-1qos78r"
          data-testid="step-button-1"
          disabled=""
          type="button"
        >
          1
        </button>
        <button
          class="chakra-button css-1qos78r"
          data-testid="step-button-2"
          disabled=""
          type="button"
        >
          2
        </button>
      </div>
      <p
        class="chakra-text css-1wvvwig"
      >
        Let's start the savings...
      </p>
    </div>
    <div
      class="css-0"
    >
      <div
        class="css-6z87o"
      >
        <p
          class="chakra-text css-18cdwwf"
        >
          Tell us a little about yourself
        </p>
        <p
          class="chakra-text css-ut9id6"
        >
          To help you get the most out of your savings and personalise this expereinces for please provide us with a few details to really help you maximise your savings and help you get to a brighter fanciancial future!
        </p>
        <div
          class="css-v8qn0c"
        >
          <div
            class="css-1d1yckv"
          >
            <label
              class="chakra-form__label css-wuh3rg"
            >
              First Name
            </label>
            <input
              class="chakra-input css-1ttok2p"
              data-testid="input-First Name"
              value=""
            />
            <p
              class="chakra-text css-80ht2y"
            />
          </div>
          <div
            class="css-1d1yckv"
          >
            <label
              class="chakra-form__label css-wuh3rg"
            >
              Last Name
            </label>
            <input
              class="chakra-input css-1ttok2p"
              data-testid="input-Last Name"
              value=""
            />
            <p
              class="chakra-text css-80ht2y"
            />
          </div>
          <div
            class="css-1d1yckv"
          >
            <label
              class="chakra-form__label css-wuh3rg"
            >
              Email
            </label>
            <input
              class="chakra-input css-1ttok2p"
              data-testid="input-Email"
              value=""
            />
            <p
              class="chakra-text css-80ht2y"
            />
          </div>
        </div>
      </div>
      <div
        class="css-j4xgao"
      >
        <button
          class="chakra-button css-1mi4eb0"
          data-testid="next-button"
          type="button"
        >
          Next
        </button>
      </div>
    </div>
  </div>
</div>
`;
